packages/iov-bns/src/bnsconnection.ts

```diff
@@ -444,46 +460,68 @@ export class BnsConnection implements BcpAtomicSwapConnection {
     return Stream.merge(offers, releases).map(combiner);
   }
 
-  public async searchTx(txQuery: BcpTxQuery): Promise<ReadonlyArray<ConfirmedTransaction>> {
+  public async searchTx(
+    txQuery: BcpTxQuery,
+  ): Promise<ReadonlyArray<ConfirmedTransaction | FailedTransaction>> {
     // this will paginate over all transactions, even if multiple pages.
     // FIXME: consider making a streaming interface here, but that will break clients
     const res = await this.tmClient.txSearchAll({ query: buildTxQuery(txQuery) });
     const chainId = await this.chainId();
     const currentHeight = await this.height();
-    const mapper = ({ tx, hash, height, txResult }: TxResponse): ConfirmedTransaction => ({
-      height: height,
-      confirmations: currentHeight - height + 1,
-      transactionId: Encoding.toHex(hash).toUpperCase() as TransactionId,
-      log: txResult.log,
-      result: txResult.data,
-      ...this.codec.parseBytes(new Uint8Array(tx) as PostableBytes, chainId),
-    });
-    return res.txs.map(mapper);
+
+    return res.txs.map(
+      (txResponse): ConfirmedTransaction | FailedTransaction => {
+        const { tx, hash, height, result } = txResponse;
+        const transactionId = Encoding.toHex(hash).toUpperCase() as TransactionId;
+
+        if (result.code === 0) {
+          return {
+            height: height,
+            confirmations: currentHeight - height + 1,
+            transactionId: transactionId,
+            log: result.log,
+            result: result.data,
+            ...this.codec.parseBytes(new Uint8Array(tx) as PostableBytes, chainId),
+          };
+        } else {
+          return {
+            transactionId: transactionId,
```

Comment:
We could fit height in here as I mentioned somewhere above.
I like to provide extra context that may be useful.